### PR TITLE
flow: Add snowflake exporter

### LINKED ISSUE #3167: Flow component for snowflake integration
Bring over the static mode snowflake integration as a Flow component.

### PR DESCRIPTION
**Pull request by JonathanWamsley**
#### PR Description

- Adds a snowflake exporter to the list of supported flow integrations.

#### Which issue(s) this PR fixes

Fixes #3167 

#### Notes to the Reviewer
First PR here! Tried my best to followed the documentation.

#### PR Checklist

- [x] CHANGELOG updated
- [x] Documentation added
- [x] Tests updated

### CODE REVIEW

**Comment by rfratto** [comment]
Hi @JonathanWamsley, sorry you've been waiting for a reviewer. I'm not the right person to review this but I'll talk to the team who owns these components and help make sure you get some eyes on this soon.

File: docs/sources/flow/reference/components/prometheus.exporter.snowflake.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+The following arguments can be used to configure the exporter's behavior.
+Omitted fields take their default values.
+
+| Name           | Type     | Description                                           | Default          | Required |
+|----------------|----------|-------------------------------------------------------|------------------|----------|
+| `account_name` | `string` | The account to collect metrics for.                   |                  | yes      |
+| `username`     | `string` | The username for the user used when querying metrics. |                  | yes      |
+| `password`     | `secret` | The password for the user used when querying metrics. |                  | yes      |
+| `role`         | `string` | The role to use when querying metrics                 | `"ACCOUNTADMIN"` | no       |
```
Comment (spartan0x117) [inline docs/sources/flow/reference/components/prometheus.exporter.snowflake.md:37]:
```suggestion
| `role`         | `string` | The role to use when querying metrics.                | `"ACCOUNTADMIN"` | no       |
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -26,6 +26,7 @@ Main (unreleased)
     the targets they reference. (@captncraig, @marctc, @jcreixell)
   - `prometheus.receive_http` - receive Prometheus metrics over HTTP (e.g. from other agents). (@thampiotr)
   - `remote.vault` retrieves a secret from Vault. (@rfratto)
+  - `prometheus.integration.snowflake` collects metrics from a snowflake database (@jonathanWamsley)
```
Comment (spartan0x117) [inline CHANGELOG.md:29]:
```suggestion
  - `prometheus.exporter.snowflake` collects metrics from a snowflake database (@jonathanWamsley)
```

File: docs/sources/flow/reference/components/prometheus.exporter.snowflake.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    username     = "USERNAME"
+    password     = "PASSWORD"
+    warehouse    = "WAREHOUSE"
+}
+
+// Configure a prometheus.scrape component to collect snowflake metrics.
+prometheus.scrape "demo" {
+  targets    = prometheus.exporter.snowflake.example.targets
+  forward_to = [ /* ... */ ]
+}
```
Comment (spartan0x117) [inline docs/sources/flow/reference/components/prometheus.exporter.snowflake.md:96]: Examples should ideally be copy-and-paste-able, only needing to replace some values to work properly. Something like this should be sufficiently complete (with a formatting change as well):
```suggestion
prometheus.exporter.snowflake "example" {
  account_name = "XXXXXXX-YYYYYYY"
  username     = "USERNAME"
  password     = "PASSWORD"
  warehouse    = "WAREHOUSE"
}

// Configure a prometheus.scrape component to collect snowflake metrics.
prometheus.scrape "demo" {
  targets    = prometheus.exporter.snowflake.example.targets
  forward_to = [ prometheus.remote_write.default.receiver ]
}

prometheus.remote_write "default" {
  endpoint {
    url = "REMOTE_WRITE_URL"
  }
}
```

**Review by spartan0x117 — COMMENTED** [review]
A few comments on docs, but overall it looks good

**Comment by JonathanWamsley** [comment]
> A few comments on docs, but overall it looks good

Thanks for the quick response time @spartan0x117 ! I added your suggested changes and will keep them in mind going forward 🚀

**Pull request merged**